apigentools/config.py

```diff
@@ -61,7 +61,7 @@ def upstream_templates_dir(self):
         return self.raw_dict.get("upstream_templates_dir", self.language)
 
     def __getattr__(self, attr):
-        return self.raw_dict[attr]
```

Comment:
Any reason why you want to access the dict with indices? I took the liberty of replacing this with a `get` so that accessing an attribute of the config object doesn't raise a `KeyError`

Reply:
Yeah. Basically we wanted to distinguish whether the user explicitly provided a `null` value or no value was present at all (which might make the code behave differently). It is however true that we haven't needed that so far, so I guess I'm ok accepting it (as this is an internal thing, we can always change it if we need if for a specific feature).

Reply:
Hmm, looking at the current code, there are accesses to various values of `LanguageConfig` instances, which shouldn't just silently pass - they'd backtrace right now, but at least they wouldn't lead to unexpected behavior of having a `None` in some string etc. While I do recognize that this is not ideal and it would be better to adapt your approach, it would require larger changes that are probably out of scope of this PR. Could you add a property for this, e.g.:

```
@property
def generate_extra_args(self):
    return self.raw_dict.get("generate_extra_args", [])
```